### PR TITLE
Add tabindex and better sortBy titles to datatables

### PR DESCRIPTION
**Pull request by dgading**
https://accessibilityinsights.io/info-examples/web/scrollable-region-focusable/

Since the datatables can be scrollable we should have a tab index set on them so keyboard navigation and focus and scroll them.

### CODE REVIEW

File: src/components/DataTable/index.jsx [hunk 1]
```diff
@@ -159,48 +159,53 @@ const DataTable = ({
   }, [filters]);
 
   return (
-    <div {...getTableProps()} className={className}>
+    <div {...getTableProps()} className={className} tabIndex="0">
```
Comment (dgading) [inline src/components/DataTable/index.jsx:162]: Adds tabindex to the table so keyboard navigation can scroll.

File: src/components/DataTable/index.jsx [hunk 2]
```diff
@@ -159,48 +159,53 @@ const DataTable = ({
...
+              {...headerGroup.getHeaderGroupProps()}
+              className={headerGroupClassName}
+            >
+              {headerGroup.headers.map((column) => {
+                return(
+                  <div key={column.id}>
+                    <div
+                      className={headerCellClassName}
+                      {...column.getHeaderProps(
+                        columnSort ? column.getSortByToggleProps((props) => ({...props, title: `Sort by ${column.Header}`})) : undefined
```
Comment (dgading) [inline src/components/DataTable/index.jsx:177]: Currently all table header titles just read as `Sort by` when hovered with a mouse. This update will change it to `Sort by my header name`. Which should clear confusion when looking at all the headers.

File: src/components/DataTable/index.jsx [hunk 3]
```diff
@@ -159,48 +159,53 @@ const DataTable = ({
...
-                      column.isResizing ? tableColumnIsResizing : ""
-                    }`}
-                  />
-                )}
-              </>
-            ))}
-          </div>
-        ))}
+        {headerGroups.map((headerGroup) => {
+          const { key } = headerGroup.getHeaderGroupProps()
```
Comment (dgading) [inline src/components/DataTable/index.jsx:165]: Added a key here and on 167 to fix React warning/errors of list items not having unique keys.

File: src/components/DataTable/index.jsx [hunk 4]
```diff
@@ -159,48 +159,52 @@ const DataTable = ({
...
-        ))}
+        {headerGroups.map((headerGroup) => {
+          return(
+            <div
+              {...headerGroup.getHeaderGroupProps()}
+              className={headerGroupClassName}
+            >
+              {headerGroup.headers.map((column) => {
+                return(
+                  <div key={column.id}>
```
Comment (dgading) [inline src/components/DataTable/index.jsx:172]: Added a key to clear a React error

**Review by dcgoodwin2112 — APPROVED** [review]
lgtm! Confirmed local changes and ran tests locally

**Pull request merged**